File: PolishPizza/promotions/src/main/java/pizzeria/promotions/promotion/service/PromotionService.java
package pizzeria.promotions.promotion.service;

import pizzeria.promotions.promotion.dto.request.PromotionCheckRequest;
import pizzeria.promotions.promotion.dto.request.PromotionPatchRequest;
import pizzeria.promotions.promotion.dto.request.PromotionRequest;
import pizzeria.promotions.promotion.dto.response.PromotionCheckResponse;
import pizzeria.promotions.promotion.dto.response.PromotionResponse;

import java.util.List;
import java.util.UUID;

public interface PromotionService {
    List<PromotionResponse> getAllPromotions();

    List<PromotionResponse> getActivePromotions();

    PromotionResponse getPromotionById(UUID id);

    PromotionResponse save(PromotionRequest request);

    void delete(UUID id);

    PromotionResponse update(UUID id, PromotionRequest request);

    PromotionResponse patch(UUID id, PromotionPatchRequest request);

    PromotionCheckResponse checkPromotion(PromotionCheckRequest request);
}

File: PolishPizza/admin/src/service/promotionService.ts
import { api as axiosInstance } from '../api/api';
import type { PromotionResponse, PromotionRequest } from '../types/promotion';

export const PromotionService = {
    
    getAll: async (): Promise<PromotionResponse[]> => {
        const response = await axiosInstance.get<PromotionResponse[]>('/promotion/'); 
        return response.data;
    },

    getOne: async (id: string): Promise<PromotionResponse | null> => {
        const response = await axiosInstance.get<PromotionResponse>(`/promotion/${id}`);
        return response.data;
    },
    
    update: async (id: string, data: PromotionRequest) => {
        const response = await axiosInstance.put(`/promotion/${id}`, data);
        return response.data;
    },
    
    create: async (data: PromotionRequest) => {
        const response = await axiosInstance.post(`/promotion/`, data);
        return response.data;
    },

    delete: async(id: string): Promise<void> => {
        const response = await axiosInstance.delete(`/promotion/${id}`);
        return response.data;
    }
};